stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,45 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::StridedMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
+    eigen!(copy(A), copy(B); sortby)
+end
+function eigen!(A::StridedMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:BlasFloat}
+   # NOTE: Copy of BunchKaufman's getproperty(B, :D) and getproperty(B, :L/U) in bunchkaufman.jl as 'getproperty' is not in place.
+   if B.rook
+        LUD, od = LAPACK.syconvf_rook!(B.uplo, 'C', B.LD, B.ipiv)
+    else
+        LUD, od = LAPACK.syconv!(B.uplo, B.LD, B.ipiv)
+    end
+    if B.uplo == 'U'
+        M = UnitUpperTriangular(LUD)
+        du = od[2:end]
+        # Aliasing of dl and du is not allowed for lu!(Tridiagonal(dl, diag(LUD), du) below.
+        dl = B.symmetric ? copy(du) : conj.(du)
+    else
+        M = UnitLowerTriangular(LUD)
+        dl = od[1:end-1]
+        # Aliasing of dl and du is not allowed for lu!(Tridiagonal(dl, diag(LUD), du) below.
+        du = B.symmetric ? copy(dl) : conj.(dl)
+    end
```

Comment:
Why can't this just be

```suggestion
   D = B.D
   M = B.uplo == 'U' ? B.U : B.L
```
? The code above is also not in-place; in fact, it does exactly the same as `getproperty`, so should be the same effort with much less code?

Reply:
Please see https://github.com/JuliaLang/julia/pull/50471#issuecomment-1636889413